server/aesdsocket.c

```diff
@@ -282,30 +397,55 @@ int main(int argc, char **argv)
     }
     
     init_sigaction();
-    addr_len=sizeof(addr);
+    addr_len=sizeof(client_addr);
+    
+    //create timestamp thread
+    pthread_t timestamp_thread;
+    threadArgs *timer_thread_args=(threadArgs *)malloc(sizeof(threadArgs));
+    if(timer_thread_args==NULL)
+    {
+    	syslog(LOG_ERR,"memory allocation for timer thread arguments failed");
+    }
+    else
+    {
+    	timer_thread_args->data_fd=data_fd;
+    	if(pthread_create(&timestamp_thread,NULL,time_stamp,(void *)timer_thread_args) !=0)
```

Comment:
@shruthi-thallapally I don't see a join of this one in cleanup, probably what the valgrind warning is telling you.